### PR TITLE
Aggregate errors rather than exiting reconcile loop per error

### LINKED ISSUE #2772: groups: attempt to reconcile all groups/members instead of failing on first error
Noticed while reviewing https://github.com/kubernetes/k8s.io/pull/2713

ref: https://github.com/kubernetes/k8s.io/pull/2713#discussion_r710252179

Right now the reconciler is relatively brittle and an invalid group/member can halt all groups reconciliation.  A better pattern to follow is to gather errors during reconciliation, and then fail at the end with the full list of errors.  I believe we do this with other github management tools like peribolos and label_sync

### PR DESCRIPTION
**Pull request by MadhavJivrajani**
Following up on https://github.com/kubernetes/k8s.io/pull/2713

Instead of returning a slice of errors, the aggregation utils defined by apimachienry utils is used mainly in accordance with how peribolos does it

As a slightly related fix - change format specifiers of error types in `fmt.Errorf()` from `%v` to `%w`

Fixes #2772 

/priority important-soon
/assign @spiffxp

### CODE REVIEW

File: groups/reconcile.go [hunk 1]
```diff
@@ -221,53 +222,66 @@ func NewReconciler(ctx context.Context, clientOption option.ClientOption) (*Reco
...
-			return fmt.Errorf("group has no email-id: %#v", g)
+			errs = append(errs, fmt.Errorf("group has no email-id: %#v", g))
+			continue
 		}
 
 		err := r.AdminSvc.CreateOrUpdateGroupIfNescessary(g)
 		if err != nil {
-			return err
+			errs = append(errs, err)
+			continue
```
Comment (spiffxp) [inline groups/reconcile.go:236]: I would prefer to drop the continues from here on. One bad member should not stop attempts to add the rest, etc

**Review by spiffxp — CHANGES_REQUESTED** [review]

File: groups/reconcile.go (hunk 1 above)
Comment (MadhavJivrajani) [reply groups/reconcile.go:236]: ahh yes, thanks for catching that

File: groups/service.go [hunk 2]
```diff
@@ -164,12 +169,12 @@ func (as *adminService) CreateOrUpdateGroupIfNescessary(group GoogleGroup) error
...
 				g4, err := as.client.InsertGroup(&g)
 				if err != nil {
-					return fmt.Errorf("unable to add new group %q: %v", group.EmailId, err)
+					return fmt.Errorf("unable to add new group %q: %w", group.EmailId, err)
 				}
 				log.Printf("> Successfully created group %s\n", g4.Email)
 			}
 		} else {
-			return fmt.Errorf("unable to fetch group %q: %#v", group.EmailId, err.Error())
+			return fmt.Errorf("unable to fetch group %q: %q", group.EmailId, err)
```
Comment (spiffxp) [inline groups/service.go:177]:
```suggestion
			return fmt.Errorf("unable to fetch group %q: %w", group.EmailId, err)
```

**Review by spiffxp — CHANGES_REQUESTED** [review]
One tiny fix otherwise lgtm

**Comment by MadhavJivrajani** [comment]
Thanks @spiffxp 
I've made the changes

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**